src/views/tickets/JSViewer.js

```diff
@@ -0,0 +1,94 @@
+import React, { Component } from 'react';
+
+import { Dropdown } from 'semantic-ui-react';
+
+import AceEditor from 'react-ace';
+import { js as beautify } from 'js-beautify';
+
+import 'ace-builds/src-noconflict/mode-javascript';
+import 'ace-builds/src-noconflict/theme-monokai';
+
+export default class CodeBlock extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      ready: false,
+    };
+  }
+
+  fileList = [];
+
+  fetchArtifacts() {
+    const artifactURL = `http://localhost:8080/api/tickets/${this.props.ticketID}/artifacts`;
+    try {
+      fetch(artifactURL, { method: 'GET' })
+        .then(res => res.json())
+        .then(res => {
+          const artifacts = res.fileArtifacts.filter(artifact =>
+            artifact.filename.endsWith('.js')
+          );
+          const numberOfFiles = artifacts.count;
```

Comment:
More newlines between declarations / conditionals / code would make this a bit easier to read. Otherwise, really like the use of promise chaining and functional programming here.

Reply:
I'd like to make it clearer.

Reply:
Here’s some writing about newline according to the [airbnb javascript style guide](https://github.com/airbnb/javascript/blob/master/README.md#whitespace--after-blocks).

Reply:
Thanks! Very useful, will peruse it.

Reply:
Perfect!